src/dialogs/more-info/controls/more-info-input_datetime.ts

```diff
@@ -0,0 +1,117 @@
+import { HassEntity } from "home-assistant-js-websocket";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  query,
+  TemplateResult,
+} from "lit-element";
+import { classMap } from "lit-html/directives/class-map";
+import type { HaDateInput } from "../../../components/ha-date-input";
+import type { PaperTimeInput } from "../../../components/paper-time-input";
+import { UNAVAILABLE_STATES, UNKNOWN } from "../../../data/entity";
+import { setInputDateTimeValue } from "../../../data/input_datetime";
+import type { HomeAssistant } from "../../../types";
+
+@customElement("more-info-input_datetime")
+class DatetimeInput extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public stateObj?: HassEntity;
+
+  @query("paper-time-input") private _timeInputEl?: PaperTimeInput;
+
+  @query("ha-date-input") private _dateInputEl?: HaDateInput;
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this.stateObj) {
+      return html``;
+    }
+
+    return html`
+      <div
+        class="more-info-input_datetime ${classMap({
+          "has-has_time": "has_time" in this.stateObj.attributes,
+          "has-has_date": "has_date" in this.stateObj.attributes,
+        })}"
+      >
+        ${this.stateObj.attributes.has_date
+          ? html`
+              <div>
+                <ha-date-input
+                  .label="${this.hass.localize(
+                    "ui.dialogs.helper_settings.input_datetime.date"
+                  )}"
+                  .disabled=${UNAVAILABLE_STATES.includes(this.stateObj.state)}
+                  .value=${`${this.stateObj.attributes.year}-${this.stateObj.attributes.month}-${this.stateObj.attributes.day}`}
```

Comment:
We dont handle `unknown` and `unavailble` states here?

Reply:
What do you mean with that comment? Should this GUI element not take that into account and leave that handling to the parent element?

Reply:
I mean that if the state is unavailable, we should not use these attributes as they will not be there?

Reply:
If we do not have any attributes then we will not even reach that point since the `${this.stateObj.attributes.has_date` before will prevent the rendering for the input.